Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift

```diff
@@ -704,6 +705,67 @@ class AppStoreConnectService {
             .map(Model.Profile.init)
     }
 
+    func createProfile(
+        name: String,
+        bundleId: String,
+        profileType: ProfileType,
+        certificateSerialNumbers: [String],
+        deviceUDIDs: [String]
+    ) throws -> Model.Profile {
+        let bundleIdResourceId = try ReadBundleIdOperation(
+            options: .init(bundleId: bundleId)
+        )
+        .execute(with: requestor)
+        .await()
+        .id
+
+        let deviceIds = try deviceUDIDs.compactMap {
+            try ListDevicesOperation(
+                options: .init(
+                    filterName: [],
+                    filterPlatform: [],
+                    filterUDID: [$0],
+                    filterStatus: nil,
+                    sort: nil,
+                    limit: nil
+                )
+            )
+            .execute(with: requestor)
+            .await()
+            .first?
+            .id
+        }
+
+        let certificateIds = try certificateSerialNumbers.compactMap {
+            try ListCertificatesOperation(
+                options: .init(
+                    filterSerial: $0,
```

Comment:
Likewise I think this could be a single call rather than multiple. You should be able to filter by multiple serials.

Reply:
Will skip this for now as the current `ListCertificatesOperation` doesn't support passing an array of `filterSerials`.
Don't want to make this PR too massive, If there is a problem with requesting time, I will create another PR for fixing this entirely.